sdk/azidentity/interactive_browser_credential_test.go

```diff
@@ -0,0 +1,89 @@
+// // Copyright (c) Microsoft Corporation. All rights reserved.
+// // Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/mock"
+	"github.com/Azure/azure-sdk-for-go/sdk/to"
+)
+
+func TestInteractiveBrowserCredential_GetTokenSuccess(t *testing.T) {
+	srv, close := mock.NewServer()
+	defer close()
+	srv.AppendResponse(mock.WithBody([]byte(accessTokenRespSuccess)))
+	cred, err := NewInteractiveBrowserCredential(&InteractiveBrowserCredentialOptions{Options: &TokenCredentialOptions{AuthorityHost: srv.URL()}})
+	if err != nil {
+		t.Fatalf("Unable to create credential. Received: %v", err)
+	}
+	authCodeReceiver = func(authorityHost string, tenantID string, clientID string, redirectURI string, scopes []string) (*interactiveConfig, error) {
+		return &interactiveConfig{
+			authCode:    "12345",
+			redirectURI: srv.URL(),
+		}, nil
+	}
+	tk, err := cred.GetToken(context.Background(), azcore.TokenRequestOptions{Scopes: []string{"https://storage.azure.com/.default"}})
+	if err != nil {
+		t.Fatalf("Expected an empty error but received: %v", err)
+	}
+	if tk.Token != "new_token" {
+		t.Fatal("Received unexpected token")
+	}
+}
+
+func TestInteractiveBrowserCredential_GetTokenInvalidCredentials(t *testing.T) {
+	srv, close := mock.NewServer()
+	defer close()
+	srv.SetResponse(mock.WithBody([]byte(accessTokenRespError)), mock.WithStatusCode(http.StatusUnauthorized))
+	cred, err := NewInteractiveBrowserCredential(&InteractiveBrowserCredentialOptions{ClientSecret: to.StringPtr(wrongSecret), Options: &TokenCredentialOptions{AuthorityHost: srv.URL()}})
+	if err != nil {
+		t.Fatalf("Unable to create credential. Received: %v", err)
+	}
+	authCodeReceiver = func(authorityHost string, tenantID string, clientID string, redirectURI string, scopes []string) (*interactiveConfig, error) {
+		return &interactiveConfig{
+			authCode:    "12345",
+			redirectURI: srv.URL(),
+		}, nil
+	}
+	_, err = cred.GetToken(context.Background(), azcore.TokenRequestOptions{Scopes: []string{scope}})
+	if err == nil {
+		t.Fatalf("Expected an error but did not receive one.")
+	}
+	var authFailed *AuthenticationFailedError
+	if !errors.As(err, &authFailed) {
+		t.Fatalf("Expected: AuthenticationFailedError, Received: %T", err)
+	} else {
```

Comment:
You can remove this `else` clause as the `t.Fatal()` above will terminate the test.

Reply:
Done